docs/en/stack/ml/nlp/ml-nlp-model-ref.asciidoc

```diff
@@ -36,6 +36,35 @@ refer to <<ml-nlp-overview>>.
 * https://huggingface.co/roberta-large
 * https://huggingface.co/microsoft/mpnet-base
 
+[discrete]
+=== Fill mask expected model output
+
+Fill mask is a specific kind of token classification; it is the base training task of many transformer models.
+
+For the Elastic stack's fill mask NLP task to understand the model output, it must have a specific format. It needs to
+be a float tensor with `shape(<number of sequences>, <number of tokens>, <vocab size>)`. 
+
+Here is an example with a single sequence `"The capital of [MASK] is Paris"` and with vocabulary
+`["The", "capital", "of", "is", "Paris", "France", "[MASK]"]`.
+
+Should output:
+
+[source]
+----
+ [
+   [
+     [ 0, 0, 0, 0, 0, 0, 0 ], // The
+     [ 0, 0, 0, 0, 0, 0, 0 ], // capital
+     [ 0, 0, 0, 0, 0, 0, 0 ], // of
+     [ 0.01, 0.01, 0.3, 0.01, 0.2, 1.2, 0.1 ], // [MASK]
+     [ 0, 0, 0, 0, 0, 0, 0 ], // is
+     [ 0, 0, 0, 0, 0, 0, 0 ] // Paris
+   ] 
+]
+----
+
+The predicted value here for `[MASK]` is obviously `"France"` with a score of 1.2.
```

Comment:
```suggestion
The predicted value here for `[MASK]` is `"France"` with a score of 1.2.
```

Reply:
`obviously` too snarky? :D

Reply:
Yeah, don't want people to feel bad if like me they take a moment to see which vocabulary item has the highest score